packages/ice/src/turn/protocol.ts

```diff
@@ -201,14 +213,24 @@ class TurnClient implements Protocol {
   refresh = () =>
     new PCancelable(async (r, f, onCancel) => {
       let run = true;
+      let timeoutHandle: NodeJS.Timeout | undefined;
       onCancel(() => {
         run = false;
+        if (timeoutHandle !== undefined) {
+          clearTimeout(timeoutHandle);
+        }
         f("cancel");
       });
 
       while (run) {
         // refresh before expire
-        await setTimeout((5 / 6) * this.lifetime * 1000);
+        let resolve: (value: void | PromiseLike<void>) => void = () => {};
+        const deferred = new Promise<void>((_r) => {
+          resolve = _r;
+        });
+
+        timeoutHandle = setTimeout(resolve, (5 / 6) * this.lifetime * 1000);
```

Comment:
Where does 5/6 come from?

Reply:
No clue, a werift thing? 😆

Reply:
it's essentially just saying - before the expected timeout, send a LIFETIME request to increase it

Reply:
where, at 6 / 6 it would be too late to refresh the LIFETIME value